ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/target/TargetDefinition.java

```diff
@@ -899,10 +901,24 @@ public boolean isContentEquivalent(ITargetDefinition definition) {
 				isArgsNullOrEqual(getVMArguments(), definition.getVMArguments()) && //
 				Objects.equals(getJREContainer(), definition.getJREContainer()) && //
 				Arrays.equals(getIncluded(), definition.getIncluded()) && //
-				Arrays.equals(getTargetLocations(), definition.getTargetLocations()) && //
+				unorderedArraysEqualsSerialized(getTargetLocations(), definition.getTargetLocations()) && //
 				Arrays.equals(getImplicitDependencies(), definition.getImplicitDependencies());
 	}
 
+	private static boolean unorderedArraysEqualsSerialized(ITargetLocation[] one, ITargetLocation[] two) {
+		if (one == two) {
+			return true;
+		}
+		if (one == null || two == null || one.length != two.length) {
+			return false;
+		}
+		Set<Object> oneXml = new HashSet<>(Stream.of(one).filter(l -> l != null)
+				.map(l -> Objects.requireNonNullElse(l.serialize(), l)).collect(Collectors.toSet()));
+		Set<Object> twoXml = new HashSet<>(Stream.of(two).filter(l -> l != null)
```

Comment:
Here and above the pipeline could be 'simplified'
```suggestion
		Set<Object> twoXml = new HashSet<>(Arrays.stream(two).filter(Objects::nonNull)
```

Reply:
Yep, and this exactly is one of the reasons why I don't call myself an experienced Java programmer yet, thanks!

Reply:
Yep, I also learned a lot in such reviews and still learn new things. 😃
Thanks for applying.